doc/common/Cadl-Project-Scripts.md

```diff
@@ -0,0 +1,127 @@
+# Integrating with cadl sync and generate scripts
+
+There are 2 common scripts provided for each language to be able to generate from within the language
+repo and use the remote cadl definition in the spec repo.
+
+## Prerequisites
+
+There are 4 things that these two scripts expect are set up in your language repo before they will run correctly.
+
+### cadl-location.yaml
+
+This file should live under the project directory for each service and has the following properties
+
+| Property | Description | IsRequired |
+| --- | --- | --- |
+| directory | The top level directory where the main.cadl for the service lives.  This should be relative to the spec repo root such as `specification/cognitiveservices/OpenAI.Inference` | true |
+| additionalDirectories | Sometimes a cadl file will use a relative import that might not be under the main directory.  In this case you can specify additional directories as a list to sync so that all needed files are synced. | false: default = null |
+| commit | The commit sha for the version of the cadl files you want to generate off of.  This allows us to have idempotence on generation until we opt into pointing at a later version. | true |
+| repo | The repo this spec lives in.  This should be either `Azure/azure-rest-api-specs` or `Azure/azure-rest-api-specs-pr`.  Note that pr will work locally but not in CI until we add another change to handle token based auth. | true |
+| cleanup | This will remove the TempCadlFiles directory after generation is complete if true otherwise this directory will be left to support local changes to the files to see how different changes would affect the generation. | false: default = true |
```

Comment:
Why make this a library specific configuration as opposed to just an option that can be set when calling the scripts? To me it seems like we should aways clean-up unless someone want to debug and then they can run the script with the do not clean-up option.

Reply:
The only advantage to having it as an option in the yaml file is that when we integrate these with idiomatic build commands its more natural to just use those commands and not try to look up how those wrap these scripts directly in order to replicate the calls.

For example in dotnet you would simply run `dotnet build /t:Generatecode` with or without the flag.  Very few people will need to know exactly how that tool wraps these scripts to know how to pass in the extra command line args.

If you feel strongly about this we can move it but I lean towards keeping it here for the idiomatic tool usecase.

Reply:
I don't feel strongly here but I do feel it is misplaced in individual library config files. I suppose if they only set it to do debugging and then remove it afterwards it might be fine but that seems like an odd usages pattern and most likely what will happen is someone will set it or copy/paste it from somewhere else and we will end up leaving the temp files all the time.